wemake_python_styleguide/visitors/tokenize/statements.py

```diff
@@ -256,3 +259,194 @@ def _post_visit(self) -> None:
             self._check_individual_line(
                 line_tokens, previous_line_token, next_line_token,
             )
+
+
+@final
+@dataclass
+class InconsistentComprehensionContext(object):
+    r"""Context for individual bracket enclosures (i.e. [],\{\}, or ())."""
+
+    def __init__(self) -> None:
+        r"""
+        Initializes context for a given bracket enclosure (i.e. [],\{\}, or ()).
+
+        Helper class for InconsistentComprehensionVisitor which stores context
+        for the current (potential) comprehension we are in. Combined with a
+        stack to enable support for nested comprehensions.
+
+        is_comprehension:
+        Flag is set if current clause is identified as a list comprehension.
+
+        seen_clause_in_line:
+        Flag is set when the current line already contains a clause, which
+        is either the action, each for loop, or the conditional. Starts off
+        as True to account for the action, which we don't actually visit.
+
+        seen_for:
+        Flag tracks whether we've seen a for statement within these brackets.
+        Effectively determines whether we are looking at some kind of
+        comprehension or not.
+
+        seen_for_in_line:
+        Flag tracks whether we've seen a for statement on this line. Used to
+        determine if a for...in statement has been split across multiple lines.
+
+        seen_if_in_line:
+        Flag tracks whether we've seen an if statement on this line. Used to
+        determine whether an in statement is from a for...in statement or
+        is a logical in.
+
+        seen_nl:
+        Flag for if we've seen any logical newlines, indicating this is a
+        multiline comprehension
+
+        potential_violation:
+        Flag for when we see multiple clauses in one line. Only a violation
+        if this is a multiline comprehension
+
+        reported:
+        Flag tracks whether we've already reported this violation.
+        """
+        self.seen_clause_in_line = False
+        self._is_comprehension = False
+        self._seen_for = False
+        self._seen_for_in_line = False
+        self._seen_if_in_line = False
+        self._seen_nl = False
+        self._potential_violation = False
+        self._reported = False
+
+    def handle_nl(self, token: tokenize.TokenInfo) -> None:
+        """
+        Handles logical newline character depending on context.
+
+        Sets appropriate flags to True if nl encountered inside brackets after
+        some clause, so that a single line comprehension with brackets on
+        multiple lines is still accepted.
+        """
+        self._seen_nl = self.seen_clause_in_line
+        self.seen_clause_in_line = False
+        self._seen_for_in_line = False
+        self._seen_if_in_line = False
+
+    def handle_for(self, token: tokenize.TokenInfo) -> bool:
```

Comment:
`handle_for`, `handle_if` and `handle_in` all are returning `bool`, so maybe it will be better to name them like `check_for` or `is_well_formed_for`, just to conform to the convention about such boolean methods/functions names

Reply:
Agreed, I think `check_for` would be much more relevant to their functionality.